examples/src/main/kotlin/io/ethers/examples/batchrequests/Multicall.kt

```diff
@@ -0,0 +1,69 @@
+package io.ethers.examples.batchrequests
+
+import io.ethers.abi.call.Multicall3
+import io.ethers.abi.call.aggregate
+import io.ethers.core.types.Address
+import io.ethers.core.types.BlockId
+import io.ethers.examples.gen.ERC20
+import io.ethers.providers.Provider
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import java.math.BigInteger
+
+/**
+ * Multicall example demonstrates how to aggregate multiple contract calls into Multicall3.
+ */
+class Multicall(
+    private val rpcUrl: String,
+    private val tokens: List<Address>,
+) {
+    // Init provider
+    private val provider = Provider.fromUrl(rpcUrl).unwrap()
+
+    fun run() {
+        val aggregate = Multicall3.aggregate(
```

Comment:
Let's add a comment explaining that we're nesting multiple aggregations, and that by default the calls are not allowed to revert.

Then, for one of the aggregations, add `.allowFailure()` at then end, which will allow that call to fail without reverting the top-level aggregation.